### ISSUE TITLE
Allow grouping of Torrents in the Web UI

### ISSUE THREAD

**Issue by MechMK1** [transmission/transmission#3813]
I seed quite a lot of software (OS images, etc.) and would like to group these together (and possibly hide them together) when looking at all my torrents.

I believe allowing users to create "groups", which can be expanded and collapsed in a tree-like manner would be an incredibly useful feature.

Even further, it would be nice to be able to add "rules" to these groups, like "Ignore seed limit" or "Max upload speed" - but that'd be quite advanced in comparison to simple grouping and collapsing.

**Comment by MechMK1** [comment]
@ckerr I just saw that the QT client doesn't include this either. I don't have a Mac, so I can't test Transmission there, but would it be safe to assume that this feature doesn't exist in either version of Transmission? If not, would it make sense to widen this issue to generally allow grouping of torrents?

**Comment by gmckeown** [comment]
Grouping or tagging would be a great feature. I have long-term torrents and short-term torrents. Would love to be able to hide the long-term torrents most of the time, so some way to group them and display/hide by group or tag would be really useful.

**Comment by dareiff** [comment]
If this was offered in the rpc calls (and I don't think it is?), I’d take this on!

((Grouping is offered in the Mac app, but not in GTK or QT? For lack of a better way to phrase this, does Mac’s version of a group carry on down to transmission-daemon? Or is it completely Mac-based?))

**Comment by killemov** [comment]
> If this was offered in the rpc calls (and I don't think it is?), I’d take this on!

Great! In the [RPC](https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md) this is called "labels". Good luck!

**Comment by elboletaire** [comment]
The current web ui has partial support for labels... you can create them by right-clicking on a torrent, and you'll see them nearby your torrent link. If you want to search by labels you need to use a special prefix `labels:`

![imatge](https://user-images.githubusercontent.com/153305/218270435-a911f836-280c-4e2d-9d0e-e8ecfe5f5b9d.png)

**Comment by ckerr** [comment]
> ((Grouping is offered in the Mac app, but not in GTK or QT? For lack of a better way to phrase this, does Mac’s version of a group carry on down to transmission-daemon? Or is it completely Mac-based?))

Yeah, I was not involved in that. In 20-20 hindsight it was a major mistake to accept a client-specific feature that didn't carry over to the other clients.

I would l-o-v-e it if someone wanted to migrate the macOS client's grouping code to libtransmission's labels so that everything uses the same code. Right now we have this unsteady situation of labels existing in libtransmission and groups existing only in macOS.

**Comment by killemov** [comment]
The ONLY term for labels should be labels. There could be confusion with bandwidth groups.

**Comment by frozenpandaman** [comment]
> If you want to search by labels you need to use a special prefix `labels:`

@elboletaire Thank you for this! Is there a way to escape spaces in labels, i.e. how do I search for a label which contains a space in it?

The macOS grouping feature is great and I hope it makes its way to the Web UI one day, along with some sort of color coding.

**Comment by svanhoutte** [comment]
Has there been any progress on this ?

**Comment by MechMK1** [comment]
Should this be assigned a milestone? Perhaps the "decide-after-4.0.0" or 5.0.0-beta.1 milestone?

I feel like this is a feature users would appreciate, but without assigning it a milestone, it'd be left by the wayside.

**Comment by j-peeters** [comment]
Hi all, thanks for the continued work on Transmisison. I see it has gotten a real boost.

I'm trying to get clear if the feature request above is the automatic management feature where (in macOS anyway) I can define rules with specific conditions for torrents (part of name, part of file etc) to give them a label and have them downloaded to a specific folder.

I've found this very useful feature for filing movies, applications and ebooks in the right folder without needing any manual management. 
Since I'm moving to a Unraid server I'm dearly missing this feature. It can only access the webUI and I noticed that these rules for management are not available in the WebUI. 

So I guess my question is, is here already a feature request for this? If not I will post one.
I'm sorry if I'm posting this in the wrong thread.

**Comment by killemov** [comment]
@j-peeters wrote:

> I'm trying to get clear if the feature request above is the automatic management feature where (in macOS anyway) I can define rules with specific conditions for torrents (part of name, part of file etc) to give them a label and have them downloaded to a specific folder.

Currently the labels functionality is just an array of strings that you can use for searching/filtering.

**Comment by j-peeters** [comment]
> Even further, it would be nice to be able to add "rules" to these groups
I would like to have rules based management in the WebUI, especially when using Transmission on a remote box. I'm switching from a MacMini as a server to an Unraid box and I'm dearly missing this feature. 
Do we need another feature request for this in a separate threads because I get the impression this thread now encompasses two or more?

**Comment by gh0sti** [comment]
would it be possible to have this feature added for the 4.1 release?

**Comment by MechMK1** [comment]
Is there any progress on this?
